File: src/test/java/etlmail/front/gui/preferences/SwingServerConfigurationTest.java
package etlmail.front.gui.preferences;

import org.fest.assertions.Assertions;
import org.junit.Test;

public class SwingServerConfigurationTest {
    @Test
    public void lesChampsDuFormulaireNeSontJamaisNuls() {
	SwingServerConfiguration builder = new SwingServerConfiguration();

	Assertions.assertThat(builder.getHost()).as("host").isNotNull();
	Assertions.assertThat(builder.getPort()).as("port").isNotNull();
	Assertions.assertThat(builder.getPassword()).as("password").isNotNull();
	Assertions.assertThat(builder.getUsername()).as("username").isNotNull();
    }

}

File: src/test/scala/etlmail/front/gui/preferences/SwingServerConfigurationTest.scala
package etlmail.front.gui.preferences

import org.scalatest.matchers.ShouldMatchers
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.junit.runner.RunWith

@RunWith(classOf[JUnitRunner])
class SwingServerConfigurationTest extends FlatSpec with ShouldMatchers {
  "the form's fields" should "never be null" in {
    // given
    val builder = new SwingServerConfiguration

    // then
    builder.host should be("")
    builder.port should be(25)
    builder.password should be("")
    builder.username should be("")
  }
}
